### PR TITLE
Switched from snakeCase to camelCase for API, renamed APIs and consolidated pending and completed jobs.

### PR DESCRIPTION
**Pull request by ivanmkc**
- Switched from snake_case to camelCase for API
- Renamed APIs
- Consolidated pending and completed jobs
- Fixed Dockerfile.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
         {
             "name": "Cloud Run: Deploy docker image",
             "request": "launch",
-            "command": "gcloud run deploy transcription --region=us-central1 --image=gcr.io/python-docs-samples-tests/forecasting-live-demo-backend:latest --platform=managed --cpu=2 --memory=8G --timeout=3600 --allow-unauthenticated",
```
Comment (ivanmkc) [inline .vscode/launch.json:42]: Updated name of Cloud Run service.

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,16 @@
 # Use the official lightweight Python image.
 # https://hub.docker.com/_/python
-FROM tiangolo/uvicorn-gunicorn:python3.7
+FROM tiangolo/uvicorn-gunicorn:python3.9
 
 WORKDIR app
 
-COPY ./requirements.txt .
-COPY ./main.py .
+COPY requirements.txt .
```
Comment (ivanmkc) [inline Dockerfile:7]: Added missing folders.

File: main.py [hunk 3]
```diff
@@ -31,14 +32,23 @@
 training_service_instance = forecast_job_service.ForecastJobService(
     training_registry=training_registry
 )
-training_jobs_manager_instance = forecast_job_coordinator.MemoryTrainingJobManager(
+training_jobs_manager_instance = forecast_job_coordinator.MemoryTrainingJobCoordinator(
     forecast_job_service=training_service_instance
 )
 
+origins = ["*"]
+app.add_middleware(
```
Comment (ivanmkc) [inline main.py:40]: Added CORS exemption.

File: main.py [hunk 4]
```diff
@@ -65,51 +75,37 @@ def preview_dataset(dataset_id: str):
...
-            "training_method_name": request.training_method_name,
-            "dataset": {
-                "id": request.dataset.id,
-                "icon": request.dataset.icon,
-                "display_name": request.dataset.display_name,
-            },
-            "model_parameters": request.model_parameters,
-            "prediction_parameters": request.prediction_parameters,
-            "start_time": request.start_time,
+@app.get("/forecast-job/{job_id}")
```
Comment (ivanmkc) [inline main.py:78]: Consolidated pending jobs and completed jobs.

File: main.py [hunk 5]
```diff
@@ -119,31 +115,31 @@ class ForecastJobAPIRequest(BaseModel):
...
+    trainingMethodName: str
+    datasetId: str
+    modelParameters: Optional[Dict[str, Any]] = None
+    predictionParameters: Optional[Dict[str, Any]] = None
 
 
-@app.post("/train")
-def train(
-    request: ForecastJobAPIRequest,
+@app.post("/submit-forecast-job")
```
Comment (ivanmkc) [inline main.py:124]: Renamed API.

File: main.py [hunk 6]
```diff
@@ -275,9 +273,9 @@ async def prediction(job_id: str, output_type: str):
             target_column = "forecast_value"  # Figure out how to generalize this.
 
             history_formatted, history_min_date, history_max_date = format_for_rechart(
-                group_column=job_request.model_parameters["time_series_id_column"],
-                time_column=job_request.model_parameters["time_column"],
-                target_column=job_request.model_parameters["target_column"],
+                group_column=job_request.model_parameters["timeSeriesIdentifierColumn"],
```
Comment (ivanmkc) [inline main.py:276]: Switched to camelCase.

File: models/completed_forecast_job.py [hunk 7]
```diff
@@ -1,10 +1,11 @@
-from datetime import datetime
+from datetime import datetime, timedelta
+from humanize.time import precisedelta
 from typing import Any, Dict, Optional
 
 from models import forecast_job_request
 
 
-class ForecastJobResult:
+class CompletedForecastJob:
```
Comment (ivanmkc) [inline models/completed_forecast_job.py:8]: Renamed class.

File: models/forecast_job_request.py [hunk 8]
```diff
@@ -27,3 +27,17 @@ class ForecastJobRequest:
 
     # The unique request id.
     id: str = dataclasses.field(default_factory=utils.generate_uuid)
+
+    def as_response(self) -> Dict:
```
Comment (ivanmkc) [inline models/forecast_job_request.py:31]: Reformatted API response format.

File: training_methods/bqml_training_method.py [hunk 9]
```diff
@@ -19,36 +19,44 @@ def training_method() -> str:
...
+            prediction_parameters (Dict[str, Any]): The prediction parameters.
 
         Returns:
             str: The model URI
         """
 
-        time_column = parameters.get("time_column")
-        target_column = parameters.get("target_column")
-        time_series_id_column = parameters.get("time_series_id_column")
+        time_column = model_parameters.get("timeColumn")
```
Comment (ivanmkc) [inline training_methods/bqml_training_method.py:39]: Switched to camelCase.

File: training_methods/training_method.py [hunk 10]
```diff
@@ -16,12 +16,18 @@ def training_method() -> str:
         pass
 
     @abc.abstractmethod
-    def train(self, dataset: dataset.Dataset, parameters: Dict[str, Any]) -> str:
+    def train(
+        self,
+        dataset: dataset.Dataset,
+        model_parameters: Dict[str, Any],
+        prediction_parameters: Dict[str, Any],
```
Comment (ivanmkc) [inline training_methods/training_method.py:23]: Added prediction_parameters dict.

**Review by soheilazangeneh — APPROVED** [review]
LGTM!

**Pull request merged**